### PR TITLE
Add mocks to NodeJS "timers" module

### PR DESCRIPTION
**Pull request by swenzel-arc**
#### Purpose (TL;DR) - mandatory

Fix issue #466 

#### Solution
It works like this:

Try to import `timers` module as `timersModule`. If it doesn't work, that's okay.

Upon `FakeTimers.install()`, iff we're installing on the `globalObject` and `timersModule` was successfully imported:
For each property that was mocked on the globalObject iff it also exists on `timersModule`:
Safe its name and original value from `timersModule` to a new property on the Clock object.
Then set it to the same property as the faked property on `globalObject`.

Upon `FakeTimers.uninstall()`, iff the object holding `timersModule`'s originals is truthy:
For each entry set the corresponding property back to the original.

### CODE REVIEW

File: test/issue-466-test.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
+"use strict";
```
Comment (fatso83) [inline test/issue-466-test.js:1]: Stuff the tests inside of this into a suitably named new describe block in `fake-timers-test.js`. We usually reserve the `issues*` files for regression test cases.

File: test/issue-466-test.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+const { sinon } = require("@sinonjs/referee-sinon");
+var timersModule;
+
+if (typeof require === "function" && typeof module === "object") {
+    try {
+        timersModule = require("timers");
+    } catch (e) {
+        // ignored
+    }
+}
```
Comment (fatso83) [inline test/issue-466-test.js:13]: Just use Mocha's built-ins to selectively run it in supported environments
```javascript
        before(function () {
            if (!timersModule) {
                this.skip();
            }
        });
```
Enables a bit better reporting :)

File: src/fake-timers-src.js [hunk 3]
```diff
@@ -886,6 +894,14 @@ function withGlobal(_global) {
                     }
                 }
             }
+            if (clock.timersModuleMocks) {
```
Comment (fatso83) [inline src/fake-timers-src.js:897]: They are not [actual mocks](https://www.martinfowler.com/articles/mocksArentStubs.html), though, so could trim that bit off.

To be consistent with the `methods` prop, just name it `timersModuleMethods`

**Review by fatso83 — CHANGES_REQUESTED** [review]
I think you could probably afford adding a line to the README that we touch the `"timers"` module, but otherwise this generally looks fine! Just a few minor things.

**Comment by benjamingr** [comment]
> Btw. do you think we should add a config flag that allows to skip altering the timers module?

No but that's a good point, it should respect the regular "should I mock setTimeout/setInterval or whatever" option (toFake)

**Review by fatso83 — APPROVED** [review]

**Comment by fatso83** [comment]
Out as 10.2.0

**Comment by fatso83** [comment]
Seems like #470 made a good candidate for why this could have been behind a flag. Some weird regression hitting us.

**Pull request merged**